src/Twig/Extension/AuthRuntime.php

```diff
@@ -191,6 +191,54 @@ public function getProviders($guid = null)
     }
 
     /**
+     * Return an array of registered OAuth providers for an account with
+     * lst steen and last updated dates
+     *
+     * @param string $guid
+     *
+     * @return array
+     */
+    public function getProvidersDates($guid = null)
+    {
+        $providers = [];
+        if ($guid === null) {
+            $auth = $this->session->getAuthorisation();
+
+            if ($auth === null) {
+              return $providers;
+            }
+            $guid = $auth->getGuid();
+        }
+
+        $providerEntities = $this->records->getProvisionsByGuid($guid);
+        if ($providerEntities === false) {
+            return $providers;
+        }
+
+        /** @var Storage\Entity\Provider $providerEntity */
+        $providers['lastseen'] = null;
+        $providers['lastupdate'] = null;
+        foreach ($providerEntities as $providerEntity) {
+            $provider['name'] = $providerEntity->getProvider();
+            $provider['lastseen'] = $providerEntity->getLastSeen();
+            $provider['lastupdate'] = $providerEntity->getLastUpdate();
+
+            $providers[] = $provider;
+            if($providers['lastseen'] <= $provider['lastseen']) {
+              $providers['lastseen'] = $provider['lastseen'];
+            }
+            if($providers['lastupdate'] <= $provider['lastupdate']) {
+              $providers['lastupdate'] = $provider['lastupdate'];
+            }
+        }
+        if($providers['lastseen'] <= $provider['lastupdate']) {
+          $providers['lastseen'] = $provider['lastupdate'];
```

Comment:
If your account is new and only logged in once the last seen is null, but last update is not.

The reason I'm looking in tho this is because a client wants to purge accounts that are unactive for more than 2 years. For this purpose the lastupdate is sufficient.

But the code might be better as follows:

```php
        // make the last seen date the same as the last update date is empty
        if (empty($providers['lastseen']) && !empty($provider['lastupdate'])) {
            $providers['lastseen'] = $provider['lastupdate'];
        }
```